[BE] Apply for a job

nspc97 opened the pull request:
[BE] Apply for a job
https://trello.com/c/E1dTAfCz/50-be-apply-for-a-job

Commit:
[BE] Apply for a job
https://trello.com/c/E1dTAfCz/50-be-apply-for-a-job

SpartakusMd commented on src/Repository/ActivityUserRepository.php:
```diff
@@ -18,4 +22,23 @@ public function __construct(RegistryInterface $registry)
+        $activityUser->setUser($applier);
+        $activityUser->setType(1);
+        $manager = $this->getEntityManager();
+        $manager->persist($activityUser);
```
Add the method `save()` to the repository and use it here.

SpartakusMd commented on src/Repository/ActivityUserRepository.php:
```diff
@@ -18,4 +22,23 @@ public function __construct(RegistryInterface $registry)
+        $activityUser = new ActivityUser();
+        $activityUser->setActivity($activity);
+        $activityUser->setUser($applier);
+        $activityUser->setType(1);
```
Use the constant from `ActivityUser` instead of number.

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -403,4 +404,53 @@ public function editAction(Activity $activity, ActivityRepository $activityRepos
+    public function applyForActivity(Activity $activity, ActivityUserRepository $repo): JsonResponse
+    {
+        $applierUser = $this->getUser();
+        $repo->apply($activity, $applierUser);
```
Before `apply()` check if user has already applied. He can't apply twice to the same activity.

SpartakusMd replied:
Also check status and deadline.

Commit:
[BE] Apply for a job
https://trello.com/c/E1dTAfCz/50-be-apply-for-a-job

Can't apply twice to the same activity

Commit:
[BE] Apply for a job
https://trello.com/c/E1dTAfCz/50-be-apply-for-a-job

Documentation annotations fixed.

Commit:
[BE] Apply for a job
https://trello.com/c/E1dTAfCz/50-be-apply-for-a-job

The owner of activity can't apply

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -403,4 +405,78 @@ public function editAction(Activity $activity, ActivityRepository $activityRepos
+     * )
+     * @SWG\Response(
+     *     response="400",
+     *     description="Already applied/ activity finnished/ deadline passed!",
```
For each distinct error you should have a separate `@SWG\Response` block.
